devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1533,6 +1629,25 @@ bool HumanStateProvider::impl::getLinkTransformFromInputData(
     return true;
 }
 
+bool HumanStateProvider::impl::computeRelativeTransformForInputData(
```

Comment:
This function is not clear to me in what it does. It takes pose of links in inertial frames and converts it to pose in base frame. Why are we doing this? and why are we modifying the same container that is storing the former into latter?

Reply:
This is done in case the following two conditions are verified:
- fixed base is used
- a sensor measurement associated with the base is present

This is the case for example, in which we have a sensor attached to the base frame, but we want to solve the fixed-base inverse kinematics. In this case, in fact, we have to consider only the relative rotation between the other sensors and the base sensors. Please let me know if that is clear.

I will also change it in order to have input-output transforms.

Reply:
> I will also change it in order to have input-output transforms.

Actually, I noticed we modify the same container also in other points. Do you think that is kind of bad, or is it fine? Since some of those methods are optional, it is kind of convient to use always the same container since we avoid to instantiate multiple variable.

Reply:
> This is done in case the following two conditions are verified:
> - fixed base is used
> - a sensor measurement associated with the base is present
> This is the case for example, in which we have a sensor attached to the base frame, but we want to solve the fixed-base inverse kinematics. In this case, in fact, we have to consider only the relative rotation between the other sensors and the base sensors. 

maybe we could add what you wrote here as comments/documentation. That should be helpful for the future.

> Do you think that is kind of bad, or is it fine? Since some of those methods are optional, it is kind of convient to use always the same container since we avoid to instantiate multiple variable.

It might seem convenient at the moment, but there is a possibility that it might come back to bite us in the future, in  case we engage in some rigorous debugging. That's my only concern. 
I see your point as well because In the other case, I think it's harder to track two different containers.

if you believe this will not become a big issue for the future, then we can proceed as it is.

Reply:
Ok, yes, for the time being, I would leave it as it is.

Meanwhile, I have added a comment that documents what I wrote above on the usage of this method.